src/vcpkg/cmakevars.cpp

```diff
@@ -318,6 +344,51 @@ endfunction()
                 block_end = std::find(block_start, port_end, BLOCK_END_GUID);
             }
 
+            if (opt_triplet_hashes)
+            {
+                auto parse_variables = [port_start, port_end](StringView start_guid, auto callback) {
```

Comment:
New parser should be extracted and tested. (And maybe use ParserBase et al. but since this is parsing our own content I don't think that's strictly necessary)